pkg/process/events/listener_linux.go

```diff
@@ -0,0 +1,165 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// Copyright 2016-present Datadog, Inc.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+
+//go:build linux
+// +build linux
+
+package events
+
+import (
+	"context"
+	"errors"
+	"go.uber.org/atomic"
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"google.golang.org/grpc"
+
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/process/events/model"
+	"github.com/DataDog/datadog-agent/pkg/security/api"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// SysProbeListener collects process events using the runtime-security module in system-probe
+type SysProbeListener struct {
+	client        api.SecurityModuleClient
+	conn          *grpc.ClientConn
+	running       atomic.Value
+	connected     atomic.Value
+	retryInterval time.Duration
+	handler       EventHandler
+	wg            sync.WaitGroup
```

Comment:
nit - perhaps add some comments here?

Reply:
Done in https://github.com/DataDog/datadog-agent/pull/12099/commits/564beab9caa3c59be70329e66e06562877c63163.